From 0063b24a57f2ae53223dca7203f1729ee691623c Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Tue, 11 Feb 2025 09:52:39 +0800
Subject: [PATCH 01/14] Fix PHP 8 exit status cannot be null error

Summary:
Passing null as the first parameter ($status) to exit has been deprecated
and leads to a PHP error.

Test Plan: Run some bin scripts with PHP 8.4.

Reviewers: #blessed_committers, O1 Blessed Committers!

Subscribers: tobiaswiese, valerio.bozzolan, Matthew, Cigaryno

Differential Revision: https://we.phorge.it/D25869
---
 src/parser/argument/PhutilArgumentParser.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/parser/argument/PhutilArgumentParser.php b/src/parser/argument/PhutilArgumentParser.php
index fccb8e9da2e0..e86124e09a68 100644
--- a/src/parser/argument/PhutilArgumentParser.php
+++ b/src/parser/argument/PhutilArgumentParser.php
@@ -491,7 +491,7 @@ final class PhutilArgumentParser extends Phobject {
     if ($workflow->isExecutable()) {
       $workflow->setArgv($this);
       $err = $workflow->execute($this);
-      exit($err);
+      exit($err ?? 0);
     } else {
       return $workflow;
     }

base-commit: 04e3e250f7da6a55c908d373a0df9949eeea6d7b
-- 
2.48.1

